Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ILLLagrange.py

```diff
@@ -0,0 +1,308 @@
+                        else:
+                            monitor_counts_col = self.MONITOR_COUNT_COL_DEFAULT
+                            self.log().warning("Could not find monitor counts column in file {}. "
+                                               "Defaulting to column {}".format(file, self.MONITOR_COUNT_COL_DEFAULT))
+                        break
+                f.close()
+
+            if header_index < 0:
+                self.log().warning("Could not determine header length in file {}. "
+                                   "Defaulting to {}".format(file, self.HEADER_SIZE_DEFAULT))
+                header_index = self.HEADER_SIZE_DEFAULT
+                energy_col = self.ENERGY_COL_DEFAULT
+                detector_counts_col = self.DETECTOR_COUNT_COL_DEFAULT
+                monitor_counts_col = self.MONITOR_COUNT_COL_DEFAULT
+
+            # load the relevant data, which position we just determined - angle, monitor count, detector count
+            data = np.loadtxt(file,
+                              dtype=float,
+                              skiprows=header_index + 2,
+                              usecols=(energy_col, monitor_counts_col, detector_counts_col))
+
+            loaded_data = np.vstack((loaded_data, data)) if loaded_data is not None else data
+
+        return loaded_data
+
+    def get_counts_and_errors(self, data):
+        """
+        Compute and return correct energy, normalized detector counts and errors
+        @param data the data to format
+        @return 3 arrays, with the values being incident energy, normalized detector counts, errors
+        """
+
+        energy = [0]*len(data)
+        detector_counts = [0]*len(data)
+        errors = [0]*len(data)
+
+        # if the user wants to see transfer energy, we subtract by the constant offset
+        offset = 0 if self.use_incident_energy else self.INCIDENT_ENERGY_OFFSET
+
+        for index, line in enumerate(data):
+            energy[index] = line[0] - offset
+
+            detector_counts[index] = line[2] / line[1]
+            errors[index] = np.sqrt(line[2]) / line[1]
+
+        return energy, detector_counts, errors
+
+    def merge_adjacent_points(self, data):
+        """
+        Merge points that are close to one another together, summing their values
+        @param data a (nb of points, 3)-shaped numpy array, with values (incident energy, monitor counts, detector counts)
+        @return a (nb of points, 3)-shaped numpy array, with data sorted and merged by their incident energy.
+        """
+        # sort by incident energy
+        data = data[data[:, 0].argsort()]
+
+        # sum adjacent data points
+        current_writing_index = 0
+        for point in data:
+            if point[0] - data[current_writing_index][0] < self.EPSILON:
+                data[current_writing_index][1] = point[1]
+                data[current_writing_index][2] = point[2]
+                # TODO decide on a merging behaviour for multiple points
```

Comment:
The behaviour here does not seem to correspond to the comment below, as the data is currently overwritten (points being removed), if I understand this correctly. There is no calculation of the mean at the moment. I presume also this is the part that awaits user's response. For now, could you correct the comment here so it better corresponds to the current behaviour?